app/api/vmapi.rb

```diff
@@ -71,7 +72,9 @@ def get_vm(name)
         host: vm.summary.runtime.host.name,
         guestHeartbeatStatus: vm.guestHeartbeatStatus,
         summary: vm.summary,
-        vmwaretools: (vm.guest.toolsStatus != 'toolsNotInstalled') }
+        vmwaretools: (vm.guest.toolsStatus != 'toolsNotInstalled'),
+        request: Request.find_by_name(vm.name)
```

Comment:
Simply finding the Request by name is not enough!
The Request must also be accepted, otherwise the Request could not have created the VM and the two are not associated!
It must also be ensured that no two Requests that are accepted have the same name, this is probably not enforced at the moment.

Reply:
We added vm_users(vm). It does not address all of these problems, but we will probably refactor the whole vmapi and fix the rest.

Reply:
we also had to rename the vmapi methods that only retrun a hash and not the real vm object. we used "_info" for the f.e. all_vm_infos.